proto/pulumi/codegen/schema/schema.proto

```diff
@@ -0,0 +1,215 @@
+// limitations under the License.
+
+syntax = "proto3";
+
+package pulumirpc.codegen;
+
+option go_package = "github.com/pulumi/pulumi/sdk/v3/proto/go/codegen/schema";
+
+// PackageInfo is a descripition of a Pulumi Package
+message PackageInfo {
+  // The unqualified name of the package (e.g. "aws", "azure", "gcp", "kubernetes", "random")
+  string name = 1;
+  // The human-friendly display name of the package
+  optional string display_name = 2;
+  // The version of the package. The version must be valid semver
+  optional string version = 3;
+  // The description of the package. Descriptions are interpreted as Markdown
+  optional string description = 4;
+  // The list of keywords associated with the package if any
+  repeated string keywords = 5;
+  // The URL to the package's homepage
+  optional string homepage = 6;
+  // The license under which the package is distributed
+  optional string license = 7;
+  // Freeform text attribution of derived work, if required
+  optional string attribution = 8;
+  // The URL to the package's repository
+  optional string repository = 9;
+  // The URL to the package's logo
+  optional string logo_url = 10;
+  // The URL to use when downloading the provider plugin binary
+  optional string plugin_download_url = 11;
+  // The name of the person or organization that published the package
+  optional string publisher = 12;
+  // Format metadata about this package
+  optional Meta meta = 13;
+  // The provider type for this package
+  optional ResourceSpec provider = 14;
+  // Additional language-specific data about the package
+  map<string, bytes> language = 15;
+  // AN optional object to define parameterization for the package
+  optional Parameterization parameterization = 16;
+  // A list of allowed package names in addition to the name property
+  repeated string allowed_package_names = 17;
+}
+
+message Parameterization {
+  // The definition of the base provider for the package
+  optional BaseProvider base_provider = 1;
+  // The parameter for the provider
+  optional bytes parameter = 2;
+}
+
+message BaseProvider {
+  // The unqualified name of the package
+  string name = 1;
+  // The version of the package. The version must be valid semver
+  string version = 2;
+}
+
+message Meta {
+  // A regex that is used by the importer to extract a module name from the module portion of
+  // a type token. Packages that use the module format \"namespace1/namespace2/.../namespaceN\"
+  // do not need to specify a format. The regex must define one capturing group that contains
+  // the module name, which must be formatted as \"namespace1/namespace2/...namespaceN\"
+  optional string module_format = 1;
+  // Write the package to support the pack command
+  optional bool support_pack = 2;
+}
+
+// A reference to the type.
+message TypeSpec {
+  // Indicates that when used as an input, this type does not accept eventual values.
+  optional bool plain = 1;
```

Comment:
Doesn't need to be optional, bool defaults to false and that's easier to reason about than wondering what true|false|none means